networkx/algorithms/tree/mst.py

```diff
@@ -1125,19 +1125,34 @@ def _write_partition(self, partition):
             A Partition dataclass describing a partition on the edges of the
             graph.
         """
-        for u, v, d in self.G.edges(data=True):
-            if (u, v) in partition.partition_dict:
-                d[self.partition_key] = partition.partition_dict[(u, v)]
-            else:
-                d[self.partition_key] = EdgePartition.OPEN
+
+        partition_dict = partition.partition_dict
+        partition_key = self.partition_key
+        G = self.G
+
+        # Iterate over edges with for multi-graphs
+        if G.is_multigraph():
+            for u, v, k, d in G.edges(keys=True, data=True):
+                edge = (u, v, k)
+                d[partition_key] = partition_dict.get(edge, EdgePartition.OPEN)
+
+        # Iterate over edges for simple graphs
+        else:
+            for u, v, d in G.edges(data=True):
+                edge = (u, v)
+                d[partition_key] = partition_dict.get(edge, EdgePartition.OPEN)
 
     def _clear_partition(self, G):
         """
         Removes partition data from the graph
         """
-        for u, v, d in G.edges(data=True):
-            if self.partition_key in d:
-                del d[self.partition_key]
+        partition_key = self.partition_key
+        edges = (
+            G.edges(keys=True, data=True) if G.is_multigraph() else G.edges(data=True)
+        )
+        for *e, d in edges:
```

Comment:
I like this pattern! What do you think about using it above when setting `d[partition_key]`? This way, we wouldn't need separate for loops for `if G.is_multigraph()` and `else`.